src/odemis/driver/test/ephemeron_test.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on 7 Feb 2014
+
+Copyright © 2014 Kimon Tsitsikas, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms
+of the GNU General Public License version 2 as published by the Free Software
+Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+"""
+import logging
+import os
+import queue
+
+import numpy
+
+from odemis.driver.ephemeron import STATE_NAME_IDLE, STATE_NAME_TRIGGER, STATE_NAME_BUSY
+from odemis import model
+from odemis.driver import ephemeron
+import time
+import unittest
+from unittest.case import skip
+
+logging.getLogger().setLevel(logging.DEBUG)
+
+# arguments used for the creation of basic components
+CONFIG_SED = {"name": "sed", "role": "sed"}
+CONFIG_BSD = {"name": "bsd", "role": "bsd"}
+CONFIG_SCANNER = {"name": "scanner", "role": "ebeam"}
+CONFIG_FOCUS = {"name": "focus", "role": "ebeam-focus"}
+CONFIG_SEM = {"name": "sem", "role": "sem", "image": "simsem-fake-output.h5",
+              "children": {"detector0": CONFIG_SED, "scanner": CONFIG_SCANNER,
+                           "focus": CONFIG_FOCUS}
+             }
+
+TEST_NOHW = (os.environ.get("TEST_NOHW", "0") != "0")  # Default to Hw testing
+
+
+class TestMightyEBICDetector(unittest.TestCase):
+    """
+    Test case to test the functionality of the driver for an Ephemeron MightyEBIC detector.
+    """
+    @classmethod
+    def setUpClass(cls):
+        if not TEST_NOHW:
+            cls.ebic_det = ephemeron.MightyEBIC("EBIC Scan Controller",
+                                                "ebic-detector",
+                                                2,
+                                                "opc.tcp://192.168.56.2:4840/mightyebic/server/",
+                                                "http://opcfoundation.org/UA/")
+        else:
+            # EBIC detector with a simulated server
+            cls.ebic_det = ephemeron.MightyEBIC("EBIC Scan Controller",
+                                                "ebic-detector",
+                                                2,
+                                                "opc.tcp://localhost:4840/freeopcua/server/",
+                                                "http://examples.freeopcua.github.io")
```

Comment:
You should change your simulator so that that these commands be as close as possible as the real hardware. So use the name URL and same namespace.
Then, the namespace can be just hard-coded in a constant.